src/main/java/emissary/core/TransformHistory.java

```diff
@@ -118,13 +135,11 @@ public List<History> getHistory() {
      * @return last place visited
      */
     @Nullable
-    public String lastVisit() {
-        List<String> historyList = get();
-        final int sz = historyList.size();
-        if (sz == 0) {
+    public History lastVisit() {
```

Comment:
They key piece here is that we no longer use the `get()` method when determining the last place we visited. That method should be reserved for use when formatting a final transform history as requested by the BaseDataObject/DropOff.